migrations/migration.go

```diff
@@ -252,18 +252,23 @@ func (m *StorageMigration) MigrateNestedValue(
 		// Read the keys first, so the iteration wouldn't be affected
 		// by the modification of the nested values.
 		var existingKeysAndValues []keyValuePair
-		dictionary.Iterate(m.interpreter, func(key, value interpreter.Value) (resume bool) {
+
+		iterator := dictionary.Iterator()
+
+		for {
+			key, value := iterator.Next(nil)
+			if key == nil {
+				break
+			}
+
 			existingKeysAndValues = append(
 				existingKeysAndValues,
 				keyValuePair{
 					key:   key,
 					value: value,
 				},
 			)
-
-			// continue iteration
-			return true
-		})
+		}
```

Comment:
I changed this, because the migration of a dictionary with a static type like `T{}` results in a panic, because the dictionary iteration function `Iterate` converts the dictionary's static type to a sema type (`IsResourceKinded`), which results in the invalid  empty intersection type `{}`.

As  dictionary iteration/migration thus no longer loads the dictionary type, `TestMigrationPanics/composite_dictionary` is therefore no longer valid/needed, and I could remove it in 8b2476c0127b5891b0c3174102ec5899331a5b2b.

Reply:
sorry a bit confused: shouldn't the migration produce a valid intersection type? or does it happens after the iteration?

Reply:
I was also confused at first 😅 Yes, the migration generates a new type for the dictionary. However, the dictionary value still has the old type while it is getting iterated over.

Reply:
aha makes sense, thanks for the explanation